### PR TITLE
fix(project): repair netlify hosting preconditions  

### PR DESCRIPTION
**Pull request by henczi-espirit**

### CODE REVIEW

File: composables/projectProperties.ts [hunk 1]
```diff
@@ -28,10 +32,21 @@ export function useProjectProperties() {
...
+      ) {
+        projectProperties = (
+          projectPropertiesResponse as NormalizedProjectPropertyResponse
+        ).projectProperties
+      }
+      projectProperties = projectPropertiesResponse as ProjectProperties
+
+      return cachedProjectProperties.value[locale] || projectProperties
     } catch (error) {
       return null
```
Comment (henczi-espirit) [inline composables/projectProperties.ts:51]: don't we need the error (with its message) here somehow?

File: composables/projectProperties.ts [hunk 2]
```diff
@@ -28,10 +32,21 @@ export function useProjectProperties() {
...
     try {
-      return (
-        cachedProjectProperties.value[locale] ||
-        (await $fsxaApi.fetchProjectProperties({ locale }))
-      )
+      const projectPropertiesResponse = await fsxaApi.fetchProjectProperties({
+        locale
+      })
+      let projectProperties = null
+      if (
```
Comment (henczi-espirit) [inline composables/projectProperties.ts:39]: We should save the result of the condition in speaking name.
like: 
const isWrappedProjectProperties = <condition>
if (isWrappedProjectProperties) { ... }

This makes it way more easy to understand the code in the future

File: composables/projectProperties.ts [hunk 3]
```diff
@@ -28,10 +32,21 @@ export function useProjectProperties() {
...
+        projectPropertiesResponse &&
+        Object.hasOwn(projectPropertiesResponse as Object, 'projectProperty')
+      ) {
+        projectProperties = (
+          projectPropertiesResponse as NormalizedProjectPropertyResponse
+        ).projectProperties
+      }
+      projectProperties = projectPropertiesResponse as ProjectProperties
+
+      return cachedProjectProperties.value[locale] || projectProperties
```
Comment (henczi-espirit) [inline composables/projectProperties.ts:49]: I think this is a bug here ->
We fetch project properties without checking the cache before

File: nuxt.config.ts [hunk 4]
```diff
@@ -35,6 +35,7 @@ export default defineNuxtConfig({
     // they need to be: NUXT_PUBLIC_<KEY> or NUXT_PRIVATE_<KEY>, where KEY is the capitalized key of the object key (logLevel -> NUXT_PUBLIC_LOG_LEVEL)
     // If you change baseUrl: process.env['NUXT_PUBLIC_SOMETHING_ELSE'], setting NUXT_PUBLIC_SOMETHING_ELSE will only work during build time
     public: {
+      baseUrl: process.env['NUXT_PUBLIC_BASE_URL'],
```
Comment (henczi-espirit) [inline nuxt.config.ts:38]: we can remove this (again)^^

File: server/api/[endpoint].post.ts [hunk 5]
```diff
@@ -1,21 +1,24 @@
...
-import { ServerErrors, FSXAProxyRoutes, FSXAApiErrors } from '~/types'
+import { FSXAApiErrors, FSXAProxyRoutes } from 'fsxa-api'
+import { ServerErrors } from '~/types'
+import { createRemoteApi } from '~/utils/fsxa'
 
 export default defineEventHandler(async (event) => {
-  const remoteApi = FSXAApiSingleton.instance // throws error if undefined
+  const runtimeConfig = useRuntimeConfig()
+  const appConfig = useAppConfig()
+  const fsxaApi = createRemoteApi(runtimeConfig, appConfig) // throws error if undefined
```
Comment (henczi-espirit) [inline server/api/[endpoint].post.ts:8]: remove comment

Comment (henczi-espirit) [reply server/api/[endpoint].post.ts:8]: todo: rename to remote api

File: utils/fsxa.ts [hunk 6]
```diff
@@ -19,7 +25,7 @@ import { LegalLink } from '~~/types'
  * @returns Dataset or null
  */
 export const fetchDatasetByRoute = async (
-  fsxaApi: FSXAProxyApi,
```
Comment (henczi-espirit) [inline utils/fsxa.ts:22]: check if we can use FSXAApi type

File: utils/tpp.ts [hunk 7]
```diff
@@ -55,10 +55,11 @@ export const waitForPreviewId = (
  * @param pageId PageId to navigate to
  */
 export const navigateToPageId = async (pageId: string) => {
-  const { $fsxaApi } = useNuxtApp()
+  const { $createContentApi } = useNuxtApp()
+  const fsxaApi = $createContentApi() as FSXAProxyApi
```
Comment (henczi-espirit) [inline utils/tpp.ts:59]: required this for testing - can we inject "createProxyApi" into tests somehow?

File: utils/tpp.ts [hunk 8]
```diff
@@ -106,19 +107,19 @@ export const onRerenderViewHandler: OnRerenderViewHandler = async () => {
   const [pageId, locale] = previewId.split('.')
 
   if (!pageId || !locale) return
-
-  const { $fsxaApi } = useNuxtApp()
+  const { $createContentApi } = useNuxtApp()
+  const fsxaApi = $createContentApi() as FSXAProxyApi
```
Comment (henczi-espirit) [inline utils/tpp.ts:111]: same as above

**Review by henczi-espirit — COMMENTED** [review]
please update

File: utils/fsxa.ts (hunk 6 above)
Comment (Dramaturco) [reply utils/fsxa.ts:22]: Yes we can

File: composables/projectProperties.ts (hunk 1 above)
Comment (Dramaturco) [reply composables/projectProperties.ts:51]: I'd suggest we make a follow-up issue of this. There has been a mistake with exporting the HttpError class of the content API. If we can fix that first, we can use it throughout the PWA-Template project for consistent error handling

**Review by Dramaturco — APPROVED** [review]

**Pull request merged**